java-frontend/src/main/java/org/sonar/java/resolve/JavaType.java

```diff
@@ -492,8 +512,14 @@ public boolean isSubtypeOf(Type superType) {
     public boolean isSubtypeOfBound(JavaType type) {
       switch (boundType) {
         case SUPER:
+          if(bound.isTagged(TYPEVAR) && !type.isTagged(TYPEVAR) && bound != type) {
```

Comment:
The 3rd Boolean expression is useless, as if `bound` is a TypeVar and `type` is not, then they are necessarily different. Moreover, this if statement could be extracted to a method, in order to use a single return statement by boundType.